src/user-has-access-react.component.tsx

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { getCurrentUser } from "@openmrs/esm-api";
+
+export default function UserHasAccessReact(props) {
+  const [user, setUser] = React.useState(null);
+
+  function hasAccess(requiredPrivilege) {
+    //@ts-ignore
+    return user.privileges.find(p => requiredPrivilege === p.display);
+  }
+
+  React.useEffect(() => {
+    const subscription = getCurrentUser({ includeAuthStatus: false }).subscribe(
+      x => {
+        setUser(x);
+      }
+    );
+    return () => {
+      subscription.unsubscribe();
+    };
+  }, []);
+
+  try {
+    if (user && hasAccess(props.privilege)) {
+      return props.children;
+    }
+  } catch (e) {}
```

Comment:
Why are we swallowing this error. And why do we even have a try/catch? I think we can totally avoid the try/catch with the following:

```js
if (user) {
  return hasAccess(props.privilege) && props.children
} else {
  return null
}
```

Reply:
done